### ISSUE TITLE
ListView Virtualization can cause items to be hidden forever if scrolling too rapidly on WASM

### ISSUE THREAD

**Issue by roxk** [unoplatform/uno#11809]
### Current behavior

1. Bind a items source of a list view to an observation collection with 200 items.
2. Launch the app, observe the first item's binding.
3. Drag scrollbar to scroll to the bottom of the list. Then scroll back up.
4. Observe the first item in the list isn't the one observed at (2).

### Expected behavior

At (4), the first item is the one observed at (2).

### How to reproduce it (as minimally and precisely as possible)

Minimal repro:
https://github.com/roxk/UnoVirtualizationWasmBug

Video:
https://user-images.githubusercontent.com/16918354/227860838-022f44aa-9d09-43c7-975e-15e7ce27afda.mp4

### Workaround

Disable virtualization by using `StackPanel` as the `ItemsPanel`. But our app currently faces severe performance issues due to this workaround.

### Works on UWP/WinUI

Yes

### Environment

Uno.UI / Uno.UI.WebAssembly / Uno.UI.Skia

### NuGet package version(s)

```
    <PackageReference Include="Microsoft.Windows.Compatibility" Version="5.0.0" />
    <PackageReference Include="Microsoft.Extensions.Logging" Version="5.0.0" />
    <PackageReference Include="Uno.Extensions.Logging.WebAssembly.Console" Version="1.4.0" />
    <PackageReference Include="Uno.UI.WebAssembly" Version="4.8.15" />
    <PackageReference Include="Uno.UI.RemoteControl" Version="4.8.15" Condition="'$(Configuration)'=='Debug'" />
    <PackageReference Include="Uno.UI.Adapter.Microsoft.Extensions.Logging" Version="4.8.15" />
    <PackageReference Include="Uno.Wasm.Bootstrap" Version="7.0.11" />
    <PackageReference Include="Uno.Wasm.Bootstrap.DevServer" Version="7.0.11" />
```

### Affected platforms

WebAssembly

### IDE

Visual Studio 2022

### IDE version

17.5.3

**Comment by roxk** [comment]
There is a similar iOS issue https://github.com/unoplatform/uno/issues/10273. Not sure if the fix is up/relevant to WASM. The issue said this is fixed in 4.6 but I can still repro in `Uno.UI.WebAssembly:4.8.15`.

**Comment by roxk** [comment]
I can reproduce in `SampleApp.Wasm`'s `VirtualizingPanelAdapterIdCache` page. Looking into it.
